main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+	"syscall"
+)
+
+const ProcessesLimit = "10"
+
+func main() {
+    if len(os.Args) < 2 {
+        panic("Few arguments")
+    }
+
+    switch os.Args[1] {
+    case "run":
+        Run(os.Args[2], os.Args[3:])
+    case "child":
+        Child(os.Args[2], os.Args[3:])
+    default:
+        panic("Bad usage")
+    }
+}
+
+// Start the parent process of the container
+func Run(command string, args []string) {
+    cmd := exec.Command("/proc/self/exe", append([]string{"child", command}, args...)...)
+    cmd.Stdin = os.Stdin
+    cmd.Stdout = os.Stdout
+    cmd.Stderr = os.Stderr
+    cmd.SysProcAttr = &syscall.SysProcAttr{
+    	Cloneflags:   syscall.CLONE_NEWUTS | syscall.CLONE_NEWPID | syscall.CLONE_NEWNS,
+        Unshareflags: syscall.CLONE_NEWNS,
+    }
+
+    must(cmd.Run())
+}
+
+// Start a child process to execute given command
+func Child(command string, args []string) {
+    fmt.Printf("Running %v with pid %d\n", append([]string{command}, args...), os.Getpid())
+
+    // Setup the new root
+    must(syscall.Sethostname([]byte("container")))
+    must(syscall.Chroot("/home/islam/Work/alpine-fs"))
+    must(os.Chdir("/"))
+    must(syscall.Mount("proc", "proc", "proc", 0, ""))
+
+    cmd := exec.Command(command, args...)
+    cmd.Stdin = os.Stdin
+    cmd.Stdout = os.Stdout
+    cmd.Stderr = os.Stderr
```

Comment:
Same comment on the `must` and the panicking applies. Also it's good practice to add a context to the error so you know exactly the error happened  on what error so for example

```go
    if err := syscall.Mount(...); err != nil {
        return fmt.Errorf("failed to mount proc: %w", err)
    }
```

Notice i use `%w` for format the error, this is a special format operator that chain the error (implements Unwrap) which means the caller can unwrap to get the original error or check for it's type.

This way when you print an error to the user it will be clear what went wrong.